public/src/components/teacher/CourseGroup.tsx

```diff
@@ -157,24 +157,23 @@ export class CourseGroup extends React.Component<ICourseGroupProps, ICourseGroup
     }
 
     private async deleteGroup(group: Group) {
-        let readyToDelete = false;
+        let readyToDelete = group.getStatus() === Group.GroupStatus.PENDING;
         const courseID = this.props.course.getId();
-        const isEmpty = await this.props.courseMan.isEmptyRepo(courseID, 0, group.getId());
-        if (!isEmpty) {
-            // if the group is pending, there is no repo, safe to delete
-            if (group.getStatus() === Group.GroupStatus.PENDING) {
-                readyToDelete = true;
-            // otherwise the group is alredy approved, ask for confimation
-            } else if (confirm(
-                `Warning! Group repository is not empty!
-                Do you still want to delete group, github team
-                and group repository?`,
-            )) {
+        // if approved group - check if repo is empty
+        if (!readyToDelete) {
+            const isEmpty = await this.props.courseMan.isEmptyRepo(courseID, 0, group.getId());
```

Comment:
We could also reuse `readyToDelete` instead of `isEmpty`. That way, we can avoid the extra `else` clause. 
```typescript
  readyToDelete = await this.props.courseMan.isEmptyRepo(courseID, 0, group.getId());
  if (!readyToDelete && !confirm(...)) {
     readyToDelete = false;
  }
}
```

Or maybe even better?:
```typescript
  readyToDelete = await this.props.courseMan.isEmptyRepo(courseID, 0, group.getId());
  if (!readyToDelete)
     readyToDelete = confirm(...)
  }
}
```


Another alternative to simplify the logic here is to do:

```typescript
readyToDelete = true;
if (!isEmpty && !confirm(...)) {
  readyToDelete = false;
}
```

Reply:
The second alternative looks best to me. I'll update after running tests on it just to be safe.

Reply:
Agree